Enhanced Review Sorting, UI Features, & Virtual Env Support

ManuLinares opened the pull request:
Improved Sorting: For score with the same rating, those with more reviews come first. (Commit: 270a453)

UI Enhancements: The search input now has autofocus by default for a smoother user experience. (Commit: 48ffe478)
  Favorite icons and checkboxes now display a cursor: pointer on hover, indicating interactivity. (Commit: 48ffe478)
  The rounded table on chords no longer displays unevenly. (Commit: 48ffe478)

Support for Virtual Environments: Added a new feature that allows users to run the project using a virtual environment (venv) and install requirements with pip. This provides a more isolated development environment and avoids dependency conflicts when not using Poetry. (Commit: d18c2e01)

Checkbox Fix: Resolved an issue where checkboxes didn't retain their status on page reloads. They are now unchecked by default. (Commit: 141a6f95)

kmille commented:
Hey
sorry for the late response. I don't like to have a `requirements.txt`. The deps are specified in `pyproject.toml`. Can you remove this part of the MR?

ManuLinares commented:
Disregard this commit entirely, the only really fix is the sorting in templates/index.html

```
diff --git a/freetar/templates/index.html b/freetar/templates/index.html
index 7672fb5..6ce3200 100644
--- a/freetar/templates/index.html
+++ b/freetar/templates/index.html
@@ -128,10 +128,17 @@
                 }
                 const arr = Array.from(th_elem.closest("table").querySelectorAll('tbody tr'))
                 arr.sort((a, b) => {
-                    const a_val = a.dataset.value || a.children[index].innerText
-                    const b_val = b.dataset.value || b.children[index].innerText
-                    return (asc) ? a_val.localeCompare(b_val) : b_val.localeCompare(a_val)
-                })
+                    const ratingA = parseFloat(a.dataset.value || a.children[index].innerText.split("/")[0]);
+                    const ratingB = parseFloat(b.dataset.value || b.children[index].innerText.split("/")[0]);
+                    const reviewsA = parseInt(a.dataset.value || a.children[index].innerText.split(" (")[1].split(")")[0]);
+                    const reviewsB = parseInt(b.dataset.value || b.children[index].innerText.split(" (")[1].split(")")[0]);
+
+                    const compareRating = ratingB - ratingA; // Higher rating comes first
+                    const compareReviews = reviewsB - reviewsA; // More reviews come first for the same rating
+
+                    // Adjust sort order based on asc flag
+                    return asc ? (compareRating || compareReviews) : -(compareRating || compareReviews);
+                });
                 arr.forEach(elem => {
                     th_elem.closest("table").querySelector("tbody").appendChild(elem)
                 })
```

ManuLinares commented:
Little time, the only fix is thie
```

diff --git a/freetar/templates/index.html b/freetar/templates/index.html
index 7672fb5..6ce3200 100644
--- a/freetar/templates/index.html
+++ b/freetar/templates/index.html
@@ -128,10 +128,17 @@
                 }
                 const arr = Array.from(th_elem.closest("table").querySelectorAll('tbody tr'))
                 arr.sort((a, b) => {
-                    const a_val = a.dataset.value || a.children[index].innerText
-                    const b_val = b.dataset.value || b.children[index].innerText
-                    return (asc) ? a_val.localeCompare(b_val) : b_val.localeCompare(a_val)
-                })
+                    const ratingA = parseFloat(a.dataset.value || a.children[index].innerText.split("/")[0]);
+                    const ratingB = parseFloat(b.dataset.value || b.children[index].innerText.split("/")[0]);
+                    const reviewsA = parseInt(a.dataset.value || a.children[index].innerText.split(" (")[1].split(")")[0]);
+                    const reviewsB = parseInt(b.dataset.value || b.children[index].innerText.split(" (")[1].split(")")[0]);
+
+                    const compareRating = ratingB - ratingA; // Higher rating comes first
+                    const compareReviews = reviewsB - reviewsA; // More reviews come first for the same rating
+
+                    // Adjust sort order based on asc flag
+                    return asc ? (compareRating || compareReviews) : -(compareRating || compareReviews);
+                });
                 arr.forEach(elem => {
                     th_elem.closest("table").querySelector("tbody").appendChild(elem)
                 })
```
